src/libstd/thread/mod.rs

```diff
@@ -713,22 +713,27 @@ struct Inner {
 
 #[derive(Clone)]
 #[stable(feature = "rust1", since = "1.0.0")]
-/// A handle to a thread.
+/// A handle to a thread, its just an abstract reference and as such
+/// it can be used to identify a thread (by name, for example). In most
+/// usage cases, this struct is not used directly.
 ///
 /// # Examples
 ///
 /// ```
 /// use std::thread;
 ///
-/// let handler = thread::Builder::new()
-///     .name("foo".into())
-///     .spawn(|| {
-///         let thread = thread::current();
-///         println!("thread name: {}", thread.name().unwrap());
+/// for i in 0..5 {
+///     let thread_name = format!("thread_{}", i);
+///     thread::Builder::new()
+///     .name(thread_name) // Now you can identify which thread panicked
+///                        // thanks to the handle's name
```

Comment:
are these lines wrapped to 80 cols? they look a bit short to me, but that could be wrong!